src/Compilers/CSharp/Test/Syntax/Parsing/ExtensionParsingTests.cs

```diff
@@ -3825,4 +3379,470 @@ public void ForVar()
         }
         EOF();
     }
+
+    [Theory, CombinatorialData]
+    public void ImplicitExtensionOperator1(bool isExplicit)
+    {
+        var text = $$"""
+            class C
+            {
+                public static {{(isExplicit ? "explicit" : "implicit")}} extension.operator int(C c) => default;
```

Comment:
this is legal syntax in C#12.  so if we're taking the route of 'record' we need to ensure it parses the same prior to the version where we add extension-type support.  We can take the break once we are in C# 13 (like how record breaks if you're on the right version of C#).